File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/tools/search/MicronautModulesSearch.py
# tag::imports[]
from jakarta.inject import Singleton
from java.util import List
from micronaut.context.annotation import Requires
from micronaut.mcp.server.tools.search import SearchRequest, SearchResponse, SearchResult, SearchTool
# end::imports[]

try:
    from io.modelcontextprotocol.common import McpTransportContext
except ImportError:  # TODO(python): packages under `io.` other than `io.micronaut` cannot be imported at runtime
    from modelcontextprotocol.common import McpTransportContext


@Requires(property="spec.name", value="MicronautModulesSearchTest")
# tag::clazz[]
@Singleton
class MicronautModulesSearch(SearchTool):

    def search(self, request: SearchRequest, transport_context: McpTransportContext) -> SearchResponse:
        return SearchResponse(List.of(SearchResult.builder()
                                      .id("micronaut-security")
                                      .title("Micronaut Security")
                                      .url("https://micronaut-projects.github.io/micronaut-security/latest/guide")
                                      .build()))
# end::clazz[]

File: test-suite-docs-java/src/test/java/io/micronaut/mcp/docs/tools/search/MicronautModulesSearch.java
package io.micronaut.mcp.docs.tools.search;

//tag::imports[]
import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.server.tools.search.SearchRequest;
import io.micronaut.mcp.server.tools.search.SearchResponse;
import io.micronaut.mcp.server.tools.search.SearchResult;
import io.micronaut.mcp.server.tools.search.SearchTool;
import io.modelcontextprotocol.common.McpTransportContext;
import jakarta.inject.Singleton;

import java.util.List;

//end::imports[]
@Requires(property = "spec.name", value = "MicronautModulesSearchTest")
//tag::clazz[]
@Singleton
class MicronautModulesSearch implements SearchTool {

    @Override
    public SearchResponse search(SearchRequest request, McpTransportContext transportContext) {
        return new SearchResponse(List.of(SearchResult.builder()
            .id("micronaut-security")
            .title("Micronaut Security")
            .url("https://micronaut-projects.github.io/micronaut-security/latest/guide")
            .build()));
    }
}
//end::clazz[]
